### ISSUE TITLE
Rule proposal: Uppercase hex code point in string

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#510]
There are 3 types of escape notation in string can have hex value, should print in uppercase, just like [number-literal-case](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/number-literal-case.md):

1. `\uXXXX` (where XXXX is 4 hex digits; range of 0x0000–0xFFFF)
2. `\u{X}` ... `\u{XXXXXX}` (where X…XXXXXX is 1–6 hex digits; range of 0x0–0x10FFFF)
3. `\xXX` (where XX is 2 hex digits; range of 0x00–0xFF)

Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String#Escape_notation

Suggest rule name `string-code-point-case`

---

Fail:

```js
const foo = "\uDeed\uBeef";
const foo = "\u{Deed}\u{Beef}";
const foo = "\xDe\xEd\xBe\xEf";
```

Pass:

```js
// escaped
const foo = "\\uDeed\\uBeef";
const foo = "\\u{Deed}\\u{Beef}";
const foo = "\\xDe\\xEd\\xBe\\xEf";

const foo = "\uDEED\uBEEF";
const foo = "\u{DEED}\u{BEEF}";
const foo = "\xDE\xED\xBE\xEF";
```

This rule should also work on `TemplateLiteral`

**Comment by sindresorhus** [comment]
How is this different from [`escape-case`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/escape-case.md)?

**Comment by fisker** [comment]
Sorry... I  didn't know it's already exits.
